Travis CI is Failing for Project

The addition of the ArrayList using System.Generics namespace in the Iterator design pattern is causing an issue.

    Behavioral/Iterator/ConcreteAggregate.cs(23,17): error CS0246: 
    The type or namespace name 'ArrayList' could not be found (are you missing a using directive or an assembly reference?) 
    [/home/travis/build/MartinVNavarro/csharp-design-patterns/DesignPatterns/DesignPatterns.csproj]

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,2 +1,8 @@
 language: csharp
 solution: csharp-design-patterns.sln
+install:
+  - nuget restore solution-name.sln
+  - nuget install NUnit.Runners -Version 2.6.4 -OutputDirectory testrunner
+script:
+  - xbuild /p:Configuration=Release solution-name.sln
+  - mono ./testrunner/NUnit.Runners.2.6.4/tools/nunit-console.exe ./MyProject.Tests/bin/Release/MyProject.Tests.dll

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,8 +1,8 @@
 language: csharp
 solution: csharp-design-patterns.sln
 install:
-  - nuget restore solution-name.sln
+  - nuget restore csharp-design-patterns.sln
   - nuget install NUnit.Runners -Version 2.6.4 -OutputDirectory testrunner
 script:
-  - xbuild /p:Configuration=Release solution-name.sln
+  - xbuild /p:Configuration=Release csharp-design-patterns.sln
   - mono ./testrunner/NUnit.Runners.2.6.4/tools/nunit-console.exe ./MyProject.Tests/bin/Release/MyProject.Tests.dll

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,4 +5,4 @@ dotnet: 2.0.0
 script:
  - dotnet restore
  - dotnet build
- - dotnet test
+ - dotnet test DesignPatters.Tests/

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,4 +5,4 @@ dotnet: 2.0.0
 script:
  - dotnet restore
  - dotnet build
- - dotnet test DesignPatters.Tests/
+ - dotnet test DesignPatterns.Tests/

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,8 +1,8 @@
 language: csharp
 solution: csharp-design-patterns.sln
-install:
-  - nuget restore csharp-design-patterns.sln
-  - nuget install NUnit.Runners -Version 2.6.4 -OutputDirectory testrunner
+mono: none
+dotnet: latest
 script:
-  - xbuild /p:Configuration=Release csharp-design-patterns.sln
-  - mono ./testrunner/NUnit.Runners.2.6.4/tools/nunit-console.exe ./MyProject.Tests/bin/Release/MyProject.Tests.dll
+ - dotnet restore
+ - dotnet build
+ - dotnet test

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,7 @@
 language: csharp
 solution: csharp-design-patterns.sln
 mono: none
-dotnet: latest
+dotnet: 1.1.5
 script:
  - dotnet restore
  - dotnet build

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,7 @@
 language: csharp
 solution: csharp-design-patterns.sln
 mono: none
-dotnet: 1.1.5
+dotnet: 2.0.0
 script:
  - dotnet restore
  - dotnet build

